aptos-move/move-examples/mint_nft/sources/mint.move

```diff
@@ -0,0 +1,124 @@
+module nft::mint {
+    use aptos_framework::account;
+    use aptos_framework::resource_account;
+    use aptos_framework::timestamp;
+    use aptos_std::ed25519;
+    use aptos_std::multi_ed25519;
+    use aptos_token::token::{Self, TokenDataId};
+    
+    use std::bcs;
+    use std::error;
+    use std::signer;
+    use std::string::{Self, String};
+    use std::vector;
+
+    struct CollectionTokenMinter has key {
+        public_key: ed25519::ValidatedPublicKey,
+        signer_cap: account::SignerCapability,
+        expiration_timestamp: u64,
+        minting_enabled: bool,
+    }
+
+    // prove that the user owns the public key and intends to acquire the NFT
+    struct MintProofChallenge has drop {
+        sequence_number: u64,
+        token_data_id: TokenDataId,
+    }
+
+    const ED25519_SCHEME: u8 = 0;
+    const MULTI_ED25519_SCHEME: u8 = 1;
+
+    const ENOT_AUTHORIZED: u64 = 1;
+    const EACCOUNT_DOES_NOT_EXIST: u64 = 2;
+    const ECOLLECTION_EXPIRED: u64 = 3;
+    const EMINTING_DISABLED: u64 = 4;
+    const EWRONG_PUBLIC_KEY: u64 = 5;
+    const EINVALID_SCHEME: u64 = 6;
+    const EINVALID_PROOF_OF_KNOWLEDGE: u64 = 7;
+
+    fun init_module(origin: &signer, collection_name: String, description: String, collection_uri: String, expiration_timestamp: u64, public_key_bytes: vector<u8>) {
```

Comment:
`init_module` is a special method that is automatically called during publish, and so only accepts a single signer argument. Since we talked about 1 of these minter contracts per contract, we can use `const` at the top for these args
(https://github.com/aptos-labs/aptos-core/blob/1f728928504a6d88fbda2b39940ccbb2e95cbea3/aptos-move/aptos-module-verifier/src/module_init.rs#L53)

Reply:
ahh i see, that's a good point

Reply:
do we have this on aptos.dev or somewhere? I was specifically looking for how `init_module` is supposed to be used but didn't find anything (but I was also only looking thru the aptos.dev docs and the move code 😅)

Reply:
examples are littered in the code base, we have yet to really kick off a docs page for move... waiting on your signer defi example to wrap up ;)

Reply:
So these are somewhat problematic... in that we don't actually have a framework that I can think of that would do what we need here.

I think this is  a good opportunity to further extend create_resource_account_and_publish...
we could add an extra thing called blob... that would be stored beside the signer cap. it can be extracted by the appropriate account. It can then use the various utils to parse the bcs fields...

let me whip something up today or tomorrow..

Reply:
https://github.com/aptos-labs/aptos-core/pull/4528/files will make this a lot easier... but we still need to figure out the right way to process the data... we could say we limit strings to being less than 128... or they blow up :P